### DISCUSSION TITLE
"Failed to deserialize workout from Peloton...."

### DISCUSSION THREAD

**Discussion by matthewhargrove** [philosowaffle/peloton-to-garmin#178] [Q&A]
I have started seeing an error message that says:

"Failed to deserialize workout from Peloton.  You can find raw data from workout here: @filename"

The workout seems to be fine, the file is there, and it uploads to Training Peaks.  Should I just ignore or is there an issue I should address?

![Untitled](https://user-images.githubusercontent.com/87453278/138961450-13b361da-badf-4816-90bb-be51a5365374.png)

**Comment by philosowaffle** [comment]
Try changing your [log level](https://github.com/philosowaffle/peloton-to-garmin/blob/master/configuration.example.json#L52) to `Debug` and lets see if we can get some more useful information on whats going on.

**Reply by matthewhargrove** [reply]
What do I change it to?  I found the code you pointed out, and it looks exactly like what you supplied, but I don't know what to change it to. Sorry, I am not familiar with coding language at all.

**Reply by philosowaffle** [reply]
Find [this line](https://github.com/philosowaffle/peloton-to-garmin/blob/master/configuration.example.json#L52) in your config, change the word `Information` to `Debug`. Restart P2G.

This config change will tell P2G to write more verbose information to the console, which can be helpful for debugging issues.

**Reply by matthewhargrove** [reply]
Thank you... here is the .json file it generated:

[workout_failed_to_deserialize_1.txt](https://github.com/philosowaffle/peloton-to-garmin/files/7428354/workout_failed_to_deserialize_1.txt)

**Reply by matthewhargrove** [reply]
and here is the on screen message:
![Untitled](https://user-images.githubusercontent.com/87453278/139122080-353cfcb3-5132-41c9-925d-ea2560f158b5.png)

**Reply by philosowaffle** [reply]
So this ended up being a bug. I've created #179 and will get a fix in shortly.
